File: 1031.maximum-sum-of-two-non-overlapping-subarrays/solution.cpp
class Solution {
public:
    int maxSumTwoNoOverlap(vector<int>& A, int L, int M) {
        int maxSum = 0;
        for (int i = 0; i < A.size(); i++) {
            for (int j = 0; j < A.size(); j++) {
                if (i + L - 1 < j && j + M - 1 < A.size() || j + M - 1 < i && i + L - 1 < A.size()) {
                    int sum = 0;
                    for (int x = i; x <= i + L - 1; x++) {
                        sum += A[x];
                    }
                    for (int x = j; x <= j + M - 1; x++) {
                        sum += A[x];
                    }
                    maxSum = max(maxSum, sum);
                }
            }
        }
        return maxSum;
    }
};

File: 1031.maximum-sum-of-two-non-overlapping-subarrays/solution.kt
import kotlin.math.max

class Solution {
    fun maxSumTwoNoOverlap(nums: IntArray, firstLen: Int, secondLen: Int): Int {
        return max(help(nums, firstLen, secondLen), help(nums, secondLen, firstLen))
    }

    private fun help(nums: IntArray, firstLen: Int, secondLen: Int): Int {
        var suml = 0
        for (i in 0 until firstLen) {
            suml += nums[i]
        }
        var maxSumL = suml
        var sumr = 0
        for (i in firstLen until firstLen + secondLen) {
            sumr += nums[i]
        }
        var res = maxSumL + sumr
        var j = firstLen
        for (i in (firstLen + secondLen) until nums.size) {
            suml += nums[j] - nums[j - firstLen]
            maxSumL = max(maxSumL, suml)
            sumr += nums[i] - nums[i - secondLen]
            res = max(res, maxSumL + sumr)
            j++
        }
        return res;
    }
}
